File: codes/SinglyLinkedList/Java/SinglyLinkedList/SinglyLinkedListMain.java
package SinglyLinkedList;

public class SinglyLinkedListMain {
    public static void main(String[] args) {
        SinglyLinkedListInterface<Integer> list = new SinlgyLinkedList<Integer>();
        Integer[] data = new Integer[1];

        list.SetSortRule((Integer d1, Integer d2) -> {
            if (d1.intValue() < d2.intValue())
                return 0;
            else
                return 1;
        });

        list.LInsert(11); list.LInsert(11);
        list.LInsert(22); list.LInsert(22);
        list.LInsert(33);

        System.out.println("현재 데이터의 수 : "+list.LCount());

        if (list.LFirst(data)) {
            System.out.print(data[0]+" ");

            while(list.LNext(data))
                System.out.print(data[0]+" ");
        }
        System.out.println();

        if (list.LFirst(data)) {
            if (data[0] == 22)
                list.LRemove();

            while (list.LNext(data))
                if (data[0] == 22)
                    list.LRemove();
        }

        System.out.println("현재 데이터의 수 : "+list.LCount());

        if (list.LFirst(data)) {
            System.out.print(data[0]+" ");

            while (list.LNext(data))
                System.out.print(data[0]+" ");
        }
        System.out.println();
    }
}

File: codes/SinglyLinkedList/Python/SinglyLinkedListMain.py
import SinglyLinkedListModule as slm


def WhoIsPrecede(d1, d2):
    if d1 < d2:
        return 0
    else:
        return 1


slist = slm.SinglyLinkedList()
data = []

slist.SetSortRule(WhoIsPrecede)

slist.LInsert(11)
slist.LInsert(11)
slist.LInsert(22)
slist.LInsert(22)
slist.LInsert(33)

print("현재 데이터의 수 : {}".format(slist.LCount()))

data = slist.LFirst()
if data[0] is True:
    print(data[1], end=' ')

    data = slist.LNext()
    while data[0] is True:
        print(data[1], end=' ')
        data = slist.LNext()
print()

data = slist.LFirst()
if data[0] is True:
    if data[1] == 22:
        slist.LRemove()
    
    data = slist.LNext()
    while data[0] is True:
        if data[1] == 22:
            slist.LRemove()
        data = slist.LNext()

print("현재 데이터의 수 : {}".format(slist.LCount()))

data = slist.LFirst()
if data[0] is True:
    print(data[1], end=' ')

    data = slist.LNext()
    while data[0] is True:
        print(data[1], end=' ')
        data = slist.LNext()
print()
